### ISSUE TITLE
Search by folder name don't work correctly when title is long

### ISSUE THREAD

**Issue by AnnyCaroline** [storybookjs/storybook#10690]
**Describe the bug**
When searching by folder's name, storybook shows the folder structure until the previous folder and don't allow to open it if the story title is long.

**To Reproduce**
1. Create a react app using `create-react-app test_storybook`
2. Add storybook using `npx -p @storybook/cli sb init`
4. Change the title of "src/stories/1-Button.stories.js" to "one/two/threelonglonglonglonglong/four/five/six/Button"
6. Open storybook and search for a folder's name

**Expected behavior**
It should show all the folder structure, like this:
![search_ok](https://user-images.githubusercontent.com/17753656/81416868-bb905200-9120-11ea-8dc5-a3f7cb3922b3.gif)

**Screenshots**
Searching by folder "six", shows folder structure until folder "five", which can't be opened.

![search_wrong](https://user-images.githubusercontent.com/17753656/81417059-fdb99380-9120-11ea-9f64-353198abd665.gif)

**Code snippets**

**src/stories/1-Button.stories.js:**
```
import React from 'react';
import { action } from '@storybook/addon-actions';
import { Button } from '@storybook/react/demo';

export default {
  title: 'one/two/threelonglonglonglonglong/four/five/six/Button',
  component: Button,
};

export const Text = () => <Button onClick={action('clicked')}>Hello Button</Button>;

export const Emoji = () => (
  <Button onClick={action('clicked')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
);
```

**System:**
Please paste the results of `npx -p @storybook/cli@next sb info` here.

Environment Info:

  System:
    OS: Linux 5.3 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
  Binaries:
    Node: 12.1.0 - ~/.nvm/versions/node/v12.1.0/bin/node
    Yarn: 1.17.3 - ~/.yarn/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v12.1.0/bin/npm
  Browsers:
    Chrome: 78.0.3904.97
    Firefox: 76.0
  npmPackages:
    @storybook/addon-actions: ^5.3.18 => 5.3.18 
    @storybook/addon-links: ^5.3.18 => 5.3.18 
    @storybook/addons: ^5.3.18 => 5.3.18 
    @storybook/preset-create-react-app: ^2.1.1 => 2.1.1 
    @storybook/react: ^5.3.18 => 5.3.18 

**Additional context**

If we search for folders before "six" or the name of the story, it works perfectly.
![search_wrong_ok](https://user-images.githubusercontent.com/17753656/81417215-36f20380-9121-11ea-82ce-2a7d4a94d8dc.gif)

If we add more folders inside "six" the error still occurs:
![search_seven](https://user-images.githubusercontent.com/17753656/81417374-7a4c7200-9121-11ea-909b-e5040c30f0cd.gif)

**Comment by shilman** [comment]
I believe this is fixed by the new 6.1 search. Can you verify @ghengeveld?

**Comment by ghengeveld** [comment]
✅ Confirmed broken in `6.0.27`, although I'm simply not getting any results at all when searching for `five` or `six`.
✅ Confirmed fixed in `6.1.0-alpha.29`.
